src/golang/lib/job/job.go

```diff
@@ -17,7 +18,7 @@ var (
 type JobManager interface {
 	Config() Config
 	Launch(ctx context.Context, name string, spec Spec) error
-	Poll(ctx context.Context, name string) (shared.ExecutionStatus, error)
+	Poll(ctx context.Context, name string, metadataPath string, storageConfig *shared.StorageConfig) (shared.ExecutionStatus, error)
```

Comment:
LambdaJobManager.Poll needs to poll the storage to check if the operator completes since there is no way to directly ask Lambda if a function is complete. The simplest solution to adding the checks for storage in the LambdaJobManager.Poll was to provide the needed info to the function. ProcessJobManager and K8sJobManager don't use these values.

Reply:
I would be more conservative if we need to change interface. Here I feel the there are two main problems
* From the interface, `Launch` and `Poll` is not symmetric. Only certain 'launch' could be polled by `poll()` using storage config and metadata path, and caller need to preserve these information.
* `Poll` is actually not technically accessing lambda but accessing other components in the system. This makes the ownership of components more convoluted.

I've introduced `operator.Poll()` in https://github.com/aqueducthq/aqueduct/pull/452/files which checks the underlying storage when some error occurred. I realized it looks like something we could easily reuse? Here we could return some error and let the operator poll do the storage checking.